hazelcast/src/main/java/com/hazelcast/aggregation/impl/BigDecimalAverageAggregator.java

```diff
@@ -57,4 +62,27 @@ public BigDecimal aggregate() {
                 BigDecimal.valueOf(count));
     }
 
+    @Override
+    public int getFactoryId() {
+        return AggregatorDataSerializerHook.F_ID;
+    }
+
+    @Override
+    public int getId() {
+        return AggregatorDataSerializerHook.BIG_DECIMAL_AVG;
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeUTF(attributePath);
+        out.writeObject(sum);
```

Comment:
BigDecimal object needs to be one of our supported serialization types. I see that it is java serializable at the moment. You may need to change this.